__tests__/ReactionEncoder.test.js

```diff
@@ -13,4 +13,44 @@ describe('ReactionEncoder class', () => {
     expect(newReaction.getReactants()).toBe(2);
     expect(newReaction.getProducts()).toBe(1);
   });
+
+  it('should be able to decode using string from idcodeexplorer', () => {
+    const idcode =
+      'gOp@DjWkB@@@ gOp@DjWkB@@@ gOp@DjWkB@@@!fhy@@@LdbbbTQQRYhTg^@@``@`@@@##!RbGwW_Wx@_c}~O}]}v`@@ !Rb@KW@gx?_`A~@M\\Bv`@@ !R|Gq~_{]||Owp?Wy?v`@@ !R_c~H?M_|uwvH_Xa}_`CW_]_|_c~H?M_|uwwW_]_|u?sZ@@@##';
+    const reaction = ReactionEncoder.decode(idcode);
+    expect(reaction.getReactants()).toBe(3);
+    expect(reaction.getProducts()).toBe(1);
+  });
+
+  it('should reencode identically', () => {
+    const idcode =
+      'gOp@DjWkB@@@ gOp@DjWkB@@@ gOp@DjWkB@@@!fhy@@@LdbbbTQQRYhTg^@@``@`@@@##!RbGwW_Wx@_c}~O}]}v`@@ !Rb@KW@gx?_`A~@M\\Bv`@@ !R|Gq~_{]||Owp?Wy?v`@@ !R_c~H?M_|uwvH_Xa}_`CW_]_|_c~H?M_|uwwW_]_|u?sZ@@@##';
+    const _reaction = ReactionEncoder.decode(idcode, {
+      ensureCoordinates: true,
+      mode:
+        ReactionEncoder.INCLUDE_DEFAULT |
+        ReactionEncoder.RETAIN_REACTANT_AND_PRODUCT_ORDER,
+    });
+
+    const baseIdCode = ReactionEncoder.encode(_reaction, {
+      keepAbsoluteCoordinates: true,
+      mode:
+        ReactionEncoder.INCLUDE_DEFAULT |
+        ReactionEncoder.RETAIN_REACTANT_AND_PRODUCT_ORDER,
+    });
+    const reaction = ReactionEncoder.decode(idcode, {
+      ensureCoordinates: true,
+      mode:
+        ReactionEncoder.INCLUDE_DEFAULT |
+        ReactionEncoder.RETAIN_REACTANT_AND_PRODUCT_ORDER,
+    });
+    const idCodeToCompare = ReactionEncoder.encode(reaction, {
+      keepAbsoluteCoordinates: true,
+      mode:
+        ReactionEncoder.INCLUDE_DEFAULT |
+        ReactionEncoder.RETAIN_REACTANT_AND_PRODUCT_ORDER,
+    });
+
+    expect(idCodeToCompare).toBe(baseIdCode);
```

Comment:
I don't understand this test. You're doing exactly the same thing twice and expecting the same result. It's technically correct, but it's not really useful. Only the use of randomness in the tested functions could end up in a failed test.

Reply:
It was to ensure decode and encode with same options had same serialization. But yes, maybe it's not the responsibility of openchemlib-js to check that.  I'll remove the test

Reply:
Can't you instead test that the idcode generated by the JS API is the same that you had at the beginning (from the java program)?

Reply:
For that, I need to decode, then re-encode.
decode and encode have params, I don't know the right combinaisons to have a working tests.

Reply:
I added a parametrized ID-Code loader into https://229-improve-web-component-wi.openchemlib-js.pages.dev/examples/generic_editor/demo page.

I should help for functional testing

Reply:
@artaius seems OK with results. to many actors interact to have stable idcode between editors (size of area, parameters of encode/decode, float precision)

So I close this conversation